include/exec/start_now.hpp

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright (c) 2021-2024 NVIDIA Corporation
+ *
+ * Licensed under the Apache License Version 2.0 with LLVM Exceptions
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *   https://llvm.org/LICENSE.txt
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include "stdexec/__detail/__execution_fwd.hpp"
+
+#include "stdexec/__detail/__concepts.hpp"
+#include "stdexec/__detail/__env.hpp"
+#include "stdexec/__detail/__receivers.hpp"
+#include "stdexec/__detail/__senders.hpp"
+#include "stdexec/__detail/__meta.hpp"
+#include "stdexec/__detail/__type_traits.hpp"
+
+namespace exec {
+  /////////////////////////////////////////////////////////////////////////////
+  // NOT TO SPEC: __start_now
+  namespace __start_now_ {
+    namespace {
+      inline constexpr auto __ref = []<class _Ty>(_Ty& __ty) noexcept {
+        return [__ty = &__ty]() noexcept -> decltype(auto) {
+          return (*__ty);
+        };
+      };
+    } // namespace
+
+    template <class _Ty>
+    using __ref_t = decltype(__ref(__declval<_Ty&>()));
+
+    struct __joiner {
+      virtual ~__joiner() {}
+      virtual void join() const noexcept {}
+    };
+
+    template <class _StgRef>
+    struct __receiver {
+      using receiver_concept = receiver_t;
+      using __t = __receiver;
+      using __id = __receiver;
+
+      using _Storage = __decay_t<__call_result_t<_StgRef>>;
+      using _Env = typename _Storage::__env_t;
+
+      _StgRef __stgref_;
+
+      template <class... _As>
+      void set_value(_As&&... __as) noexcept {
+        auto __joiner = __stgref_().__joiner_.exchange(nullptr);
+        if (__joiner) {__joiner->join();}
+      }
+
+      template <class _Error>
+      void set_error(_Error&& __err) noexcept = delete;
+
+      void set_stopped() noexcept {
+        auto __joiner = __stgref_().__joiner_.exchange(nullptr);
+        if (__joiner) {__joiner->join();}
+      }
+
+      // Forward all receiever queries.
+      auto get_env() const noexcept -> _Env {
+        return __stgref_().__env_;
+      }
+    };
+
+    static inline const __joiner __empty_joiner_{};
```

Comment:
constexpr results in: `error: constexpr variable cannot have non-literal type 'const exec::__start_now_::__joiner'` 

If I get rid of the virtual base this will be possible..